src/ripple/app/ledger/InboundLedger.cpp

```diff
@@ -1120,7 +1120,7 @@ int InboundLedger::processData (std::shared_ptr<Peer> peer,
     if ((packet.type () == protocol::liTX_NODE) || (
         packet.type () == protocol::liAS_NODE))
     {
-        std::list<SHAMapNodeID> nodeIDs;
+        std::vector<SHAMapNodeID> nodeIDs;
         std::list< Blob > nodeData;
```

Comment:
I'm a little surprised that you left `nodeData` as a list.  It looks to me like the only use of `nodeData` is when calling `takeTxNode` and `takeAsNode`.  They both are passed a `const std::list<Blob >& data`, so the `list` isn't mutated by them.  I probably missed something...